src/yang.act

```diff
@@ -55,8 +55,13 @@ extension NameRevMap[T](Collection[(str, (?str, T))]):
     def __len__(self) -> int:
         return len(self.items)
 
-def compile(yang_sources: list[str]) -> yang.schema.DRoot:
-    """Parse and compile a set of YANG modules and submodules into a schema tree
+def compile_to_dmodule(yang_sources: list[str]) -> list[yang.schema.DModule]:
```

Comment:
I think we should strive and encourage modularity and composability by pipelining. SchemaNode is the natural format for transforms etc. Conversion to DNode should only happen at the very end, so I suggest that we rename this function and let it return SchemaNode, or I guess yang.schema.Module.
```suggestion
def compile_modules(yang_sources: list[str]) -> list[yang.schema.Module]:
```

then we conversion to DNode and merge into a single root is in another function, possibly just inlined in the `compile` function